demonstrations/tutorial_implicit_diff_susceptibility.py

```diff
@@ -0,0 +1,656 @@
+learning recently - from hyperparameter optimization to training neural ordinary
+differential equations (ODEs) and even defining a whole class of new architectures
+called Deep Equilibrium Models (DEQs) [#implicitlayers]_.
+
+The idea of implicit differentiation can be applied in quantum physics to extend
+the power of automatic differentiation to situations where we are not able to
+explicitly write down the solution to a problem. As a concrete example, consider
+a variational quantum algorithm (VQA) that computes the ground-state solution of
+a parameterized Hamiltonian :math:`H(a)` using a variational ansatz
+:math:`|\psi_{z}\rangle` where :math:`z` are the variational parameters such
+that we have the solution
+
+.. math::
+
+    z^{*}(a) = \arg\,\min_{z} \langle \psi_{z}|H(a)|\psi_z\rangle.
+
+The solution changes as we change :math:`H(a)` therefore we do not have an
+explicit function as the solution :math:`z^{*}(a)`. If we are interested in properties of the
+solution state, we could use measure expectation values for some operator
+:math:`A` as
+
+.. math::
+
+    \langle A \rangle (a) = \langle \psi_{z^{*}(a)}| A | \psi_{z^{*}(a)}\rangle.
+
+With a VQA, we can find a solution to the optimization for a fixed :math:`H(a)`
+However, just like the folium of Descartes, we do not have an explicit solution
+so the gradient :math:`\partial_a \langle A \rangle (a)` is not easy to compute.
+The solution is only implicitly defined.
+
+Automatic differentiation techniques that construct an explicit computational
+graph and differentiate through it by applying the chain rule for gradient
+computation cannot be applied here easily. A brute-force application of automatic
+differentiation through the full optimization that finds :math:`z^{*}(a)`
+would require keeping track of all intermediate variables and steps in the optimization
+and differentiating through them. This could quickly become computationally
+expensive and memory intensive for quantum algorithms. Implicit differentiation
+provides an alternative way to efficiently compute such a gradient. 
+
+:math:`\partial_{a} \langle A\rangle` is the so-called 
+*generalized susceptibility* arising from condensed-matter physics. The computation
+of this quantity would otherwise require tedious analytical derivations or
+finite difference approximations. Similarly there exist various other
+interesting quantities written as gradients of a ground-state solution, 
+e.g., nuclear forces in quantum chemistry, permanent electric dipolestatic
+polarizability, the static hyperpolarizabilities of various orders,
+fidelity susceptibilities, and geometric tensors. All such
+quantities could possibly be computed using implicit differentiation on quantum
+devices. In our recent work, we present a unified way to implement such
+computations and other applications of implicit differentiation through
+variational quantum algorithms [#Ahmed2022]_.
```

Comment:
```suggestion
Similarly, there exist various other
interesting quantities that can be written as gradients of a ground-state solution, 
e.g., nuclear forces in quantum chemistry, permanent electric dipolestatic
polarizability, the static hyperpolarizabilities of various orders,
fidelity susceptibilities, and geometric tensors. All such
quantities could possibly be computed using implicit differentiation on quantum
devices. In our recent work we present a unified way to implement such
computations and other applications of implicit differentiation through
variational quantum algorithms [#Ahmed2022]_.
```
The first sentence basically restates the last sentence of the previous paragraph.